lib/elastic_apm/metrics.rb

```diff
@@ -55,7 +59,9 @@ def start
           system: CpuMemSet,
           vm: VMSet,
           breakdown: BreakdownSet,
-          transaction: TransactionSet
+          transaction: TransactionSet,
+          # jvm: JVMSet <-- we need this, but dynamically somehow
+          # maybe they register themselves akin to how the spies do?
```

Comment:
Maybe I'm missing something, but can't you just do the following?

```ruby
        if @sets.nil?
          @sets = {
            system: CpuMemSet,
            vm: VMSet,
            breakdown: BreakdownSet,
            transaction: TransactionSet
          }
          @sets[:jvm] = JVMSet if defined?(JRUBY_VERSION)
          @sets.each_with_object({}) do |(key, kls), sets|
            debug "Adding metrics collector '#{kls}'"
            sets[key] = kls.new(config)
          end
        end
```

Reply:
Sure, yes, that'll work 👍 Maybe this?

```ruby
@sets[:jvm] = JVMSet if defined?(JVMSet)
```

It's only there if JRuby already and I think it looks better, idk 🤷‍♂️